src/components/ArticlePreview.jsx

```diff
@@ -4,34 +4,40 @@ import { Link } from 'react-router';
 export default class ArticlePreview extends React.Component {
 
   render () {
-    var article = this.props.article;
+    let article = this.props.article;
 
     // Render all contributing authors
     function renderAuthors() {
       // Return nothing if no authors listed
-      return (article.authors || []).map((author) => {
-          <div>
-            <Link to={'/author/' + author.slug}>
-              {author.name}
-            </Link>
-          </div>
-      });
+      let data = [];
+      if (article.authors){
+        for (var author in article.authors){
+          if (article.authors.hasOwnProperty(author)) {
```

Comment:
Map doesn't work as expected when you are working with objects rather than arrays. For..in is safest, especially as we continue to edit the structure of posts.js because it works for both arrays and objects. 

Why do I need to check what again?